csrc/codegen.cpp

```diff
@@ -2798,6 +2827,8 @@ class CudaKernelGenerator : private OptOutConstDispatch {
   std::deque<const kir::ForLoop*> grouped_loops_;
   //! Used to replace symbolic indices with concrete values
   std::unordered_map<const Int*, int64_t> index_replacement_map_;
+  //! Keep track of thread alignment property
+  std::deque<bool> aligned_scope_exprs_;
```

Comment:
Why this is not a `vector`?

Reply:
Just to do pop and push

Reply:
I think you only used `push_back` and `pop_back`? For this case, `vector` is better?

Reply:
Isn't it better to use a linked list-like structure?

Reply:
Really? I think for vector, in most cases, a `push_back` is just a `storage[size++] = item`, and for linked-list like data structure, you will need to create a node and manipulate two pointers? I find this https://baptiste-wicht.com/posts/2012/12/cpp-benchmark-vector-list-deque.html

Reply:
Thanks. That makes sense. I don't expect there's much performance difference as it shouldn't be a bottleneck, but I'll change it to use vector.